evalml/pipelines/utils.py

```diff
@@ -77,13 +77,13 @@ def _get_preprocessing_components(
         logical_types.Integer,
     }
 
-    if len(input_logical_types.intersection(types_imputer_handles)) > 0:
-        pp_components.append(Imputer)
-
     text_columns = list(X.ww.select("NaturalLanguage", return_schema=True).columns)
     if len(text_columns) > 0:
         pp_components.append(TextFeaturizer)
 
+    if len(input_logical_types.intersection(types_imputer_handles)) > 0:
```

Comment:
I think this should be `len(input_logical_types.intersection(types_imputer_handles)) or len(text_columns)`. If there are only text features we want to also add an imputer to impute any possible missing values.